core/src/main/java/org/mule/runtime/core/internal/event/DefaultEventBuilder.java

```diff
@@ -358,6 +359,11 @@ private InternalEventImplementation(BaseEventContext context, Message message,
       this.notificationsEnabled = notificationsEnabled;
     }
 
+    private void readObject(ObjectInputStream is) throws IOException, ClassNotFoundException {
```

Comment:
I don't think this method is really necessary. If this is just about initializing the transient map, just do it in the field declaration

Reply:
I thought the same. But apparently, when java deserializes an object, it does not set transient field default values.

Reply:
On the kryo serializer part, yes, i should replicate the same behaviour. Because, if kryo ignores transient fields, it will not set the internalParameters when deserializing. On the other hand,if the approach to add a default value to the field works (maybe i am missing something), it should not need changes